### PR TITLE
Boost dependencies

### PR DESCRIPTION
**Pull request by RafaRuiz**
- Update Gradle
- Update dependencies
- Force non-vulnerable dependencies to be transitive in our tree

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,13 +10,15 @@ buildscript {
 
     ext.versions = [
             'kotlin': '1.8.0',
-            'okhttp': '3.14.7',
-            'bouncycastle': '1.61',
+            'okhttp': '4.12.0',
+            'bouncycastle': '1.69',
```
Comment (mpbarnwell) [inline build.gradle:14]:
```suggestion
```

File: api/build.gradle [hunk 2]
```diff
@@ -57,3 +56,8 @@ afterEvaluate {
         }
     }
 }
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
```
Comment (mpbarnwell) [inline api/build.gradle:63]: You've got the other modules set to 17, should this one still be 11?

**Review by mpbarnwell — APPROVED** [review]
Good start, let's merge it and see what impact it has once the main build runs.

File: build.gradle (hunk 1 above)
Comment (RafaRuiz) [reply build.gradle]: there are two suggestions, then you removed bouncycastle, but it's used. Can you clarify?

Comment (RafaRuiz) [reply build.gradle]: I've also changed it to 1.72 and `jdk18on` which doesn't have vulnerabilities

Comment (RafaRuiz) [reply build.gradle]: image

Comment (RafaRuiz) [reply build.gradle]: looks like defined globally, and I think it was done on purpose. I can separate them by module

**Pull request merged**